Add an indicator that represents the current time

jonathanweiss opened the pull request:
In the week and day view a horizontal line represents the current time.
Apple Calendar and other apps use this concept to provide the current
time as a subtle hint.

The Y position of line is calculated in the TimeGrid by using a date
diff between this.props.max and this.props.min, respectively the
diff between current date and this.props.min.
To convert the result to a pixel value, the height of the time gutter
is used as a reference.
The position of the time indicator is calulated and updated every
minute using setInterval().

Please let me know if you want this feature to be toggled using a
prop.

![screen shot 2016-08-11 at 18 03 50](https://cloud.githubusercontent.com/assets/39352/17595608/fc51e6ea-5fed-11e6-8d5b-eb16db4d28e4.png)

jonathanweiss commented:
I improved the indicator by moving it away from the time gutter and adding a little dot to the beginning of it.
![screen shot 2016-08-15 at 17 25 17](https://cloud.githubusercontent.com/assets/39352/17669285/636efd46-630d-11e6-812a-a8596e4e8ce8.png)

jquense commented:
thanks for the PR! Looks good, tho maybe the dot needs a bit more positioning? Can you also add some shots of what it looks like amidst events (with overlay)?

jonathanweiss commented:
I changed the dot from using an unicode char to a div with border-radius. That should work better without having to worry about font-rendering in different browser. Also the indicator overlaps events now (like in iCal).
![screen shot 2016-08-17 at 17 04 02](https://cloud.githubusercontent.com/assets/39352/17741592/135d0000-649d-11e6-8013-50eec69a9473.png)
![screen shot 2016-08-17 at 17 04 17](https://cloud.githubusercontent.com/assets/39352/17741593/136421dc-649d-11e6-8901-c1159825ac6f.png)

jonathanweiss commented:
I'm also working on an extension for the indicator that displays the current time, but I need to polish the code a bit  more before I'll create a different PR for that.
![screen shot 2016-08-17 at 17 13 01](https://cloud.githubusercontent.com/assets/39352/17741829/02ae0802-649e-11e6-8587-0e9e9dea9f40.png)

jquense commented:
nice! I think tho I like the events overlapping the indicator, seems less than ideal to have it obscure information

jquense commented on src/TimeGrid.jsx:
```diff
@@ -74,6 +74,17 @@ export default class TimeGrid extends Component {
+    this.positionTimeIndicator();
+
+    // Update the position of the time indicator every minute
+    this._timeIndicatorInterval = window.setInterval(() => {
```
better to use a recursive `setTimeout` here, setInterval can "drift" depending on how clogged the JS event loop is, which can cause it to update rather closely together vs actually every minute. `setTimeout` will ensure that its not called more than once every minute

jonathanweiss replied:
I'm not to concerned about an interval that is triggered every second. Using recursive `setTimeout()` we might exceed the stack trace of the browser after having the calendar open for ~six hours. ;-)
Anyway, I did change to code to work with `setTimeout()`.

I also noticed that I forgot to check for the min time frame (`this.props.min`) and added this check.

The indicator is now rendered **behind** the event.

jonathanweiss commented:
FYI: I changed the order of the events and time indicator so that the time indicator is rendered behind events and switched to `setTimeout()`.
The current time next to the indicator will be done in another pull request when I have the time to improve my current solution...

jquense commented:
thanks a ton!
